backend/src/tests/tests.test.js

```diff
@@ -0,0 +1,191 @@
+const { createTestClient } = require('apollo-server-testing');
+const { ApolloServer, gql } = require("apollo-server");
+const {beforeEach, describe, expect, it} = require("@jest/globals");
+const PostsAPI = require("../datasources/postApi");
+const UsersAPI = require("../datasources/userApi");
+const typeDefs = require('../typeDefs');
+const resolvers = require('../resolvers');
+
+let usersApi = new UsersAPI();
+let postsApi = new PostsAPI();
+
+const server = new ApolloServer({
+    typeDefs, 
+    resolvers, 
+    dataSources: () => {
+        return {
+          usersApi: usersApi,
+          postsApi: postsApi
+        }
+    }
+});
+
+const { query, mutate } = createTestClient(server);
+
+describe("query", () => {
+
+    it("posts with nested user object", async () => {
+        const GET_POSTS = gql`
+            query {
+                posts {
+                    title 
+                    author {
+                        name
+                    }
+                }
+            }
+        `;
+
+        const {
+            data: { posts }
+        } = await query({ query: GET_POSTS });
+
+        expect(posts).toHaveLength(2);
+        expect(posts).toEqual([
+            {
+                title: 'The Thing',
+                author: {
+                    name: "Peter"
+                },
+            },
+            {
+                title: 'The Nothing',
+                author: {
+                    name: "Peter"
+                },
+            }
+        ]);
+    });
+
+    it("users with nested post object", async () => {
+        const GET_USERS = gql`
+            query {
+                users {
+                    name 
+                    posts {
+                        title
+                        votes
+                    }
+                }
+            }
+        `;
+
+        const {
+            data: { users }
+        } = await query({ query: GET_USERS });
+
+
+        expect(users).toHaveLength(2);
+        expect(users).toEqual([
+            {
+                name: "Peter",
+                posts: [
+                    {
+                        title: "The Thing",
+                        votes: 0
+                    },
+                    {
+                        title: "The Nothing",
+                        votes: 0
+                    }
+                ]
+            }, 
+            {
+                name: "Max",
+                posts: []
+            }
+        ]);
+    });
+  
+});
+
+describe("write(post: $postInput)", () => {
+
+    let WRITE_POST;
+
+    beforeEach(() => {
+        WRITE_POST = gql`
```

Comment:
```suggestion
    let WRITE_POST = gql`
```
This doesn't get mutated in your test cases so no need to re-initialize.